hadoop-maven-plugins/src/main/java/org/apache/hadoop/maven/plugin/protoc/ProtocRunner.java

```diff
@@ -172,6 +172,10 @@ private long computeChecksum(File file) throws IOException {
 
     public void writeChecksums() throws IOException {
       ObjectMapper mapper = new ObjectMapper();
+      boolean folderCreated = checksumFile.getParentFile().mkdirs();
+      if (!folderCreated) {
+        throw new IOException("Unable to create path");
+      }
```

Comment:
If the parent already exists and it is directory, `folderCreated` will be false and throw IOException. We need to first check if the parent is directory.

Reply:
@aajisaka - The other way I can I can think of will be directly doing

```
checksumFile.getParentFile().mkdirs();
```

and not checking the results at all. But then SpotBug will complain about the result not checked.

Reply:
@aajisaka  - Shall I go ahead and do it this way and use @SuppressWarnings tag along?

Reply:
Other option I can think is 

```
File parent = checksumFile.getParentFile();
if (parent != null && !parent.exists() && !parent.mkdirs()) {
        throw new IOException("Couldn't create dir: " + parent);
}
```

Reply:
@aajisaka - Please share your advice. Thanks.

Reply:
I prefer to use NIO.2 library `Files.createDirectories(checksumFiles.getParentFile().toPath())`.

Reply:
> I prefer to use NIO.2 library `Files.createDirectories(checksumFiles.getParentFile().toPath())`.

Thanks for your suggestion. I have made the changes, can you please review it again.